### PR TITLE
Switch from TravisCI to Github Action workflows

### PR DESCRIPTION
**Pull request by alexmisk**
Hi!

I've tried to rewrite your Travis CI workflow with Github Actions but your feedback needed.

**Current status:**
- there are two workflows:
	- test.yml workflow
		- fires on every push to non-master branch and every pull request
		- runs tests on multiple python versions in parallel
		- runs benchmark if all tests are passed
		- checks coverage if all tests are passed
	- build.yml
		- fires only if master branch was updated
		- do everything test.yml does + build docs and updates gh-pages branch (only if all tests are passed also)
- tests run smoothly on all defined Python versions (3.6 - 3.9, pypy3)
- coverage check and benchmark run are decoupled into two separate steps for better maintainability
- gh-pages update is now slightly more elegant thanks to "GitHub Pages Deploy Action" (https://github.com/JamesIves/github-pages-deploy-action)

**Drawbacks:**
- workflows are not optimised yet in terms of speed (no cache etc.)
- same codeblocks are used in test.yml and build.yml. This is avoidable, though. If you will never push directly to master branch you can safely remove 'test', 'check_coverage' and 'run_benchmark' steps from build.yml
- gh-pages commit message is always "docs: Update docs". I beleive this is also fixable
- .travis.yml should be removed because it will push changes in gh-pages branch instead

**Questions:**
1. Is decoupling coverage check from brnchmark OK for you?
2. `make coverage` also run tests. This is usable during local developement, but slow things up in CI. Maybe you can add a separate Makefile command just for checking coverage in CI?
3. Do you absolutely need gh-pages commit messages to be verbose (with a reference to commit or pull request)?

Please review the PR and let me know if anything else needed.

### CODE REVIEW

**Comment by yukinarit** [comment]
Wow. Great work!

I am still new to Github Actions so please allow me to answer your questions little by little. 

> same codeblocks are used in test.yml and build.yml. This is avoidable, though. If you will never push directly to master branch you can safely remove 'test', 'check_coverage' and 'run_benchmark' steps from build.yml

Your CI scripts look neat however I also feel like to remove the repeating code block. I won't push to master directly so I don't mind removing `test`, `check_coverage` and `run_benchmark` from build.yml 👍

**Comment by yukinarit** [comment]
> make coverage also run tests. This is usable during local developement, but slow things up in CI. Maybe you can add a separate Makefile command just for checking coverage in CI?

Yes, I don't mind separating coverage and test in Makefile. It's now just written as it works well on my local and travis 😄 

> Do you absolutely need gh-pages commit messages to be verbose (with a reference to commit or pull request)?

Yeah, it was supposed to embed PR number (or URL) in the commit title but it seems not working correctly.. I absolutely want more information in gh-pages commits

**Comment by alexmisk** [comment]
> so I don't mind removing `test`, `check_coverage` and `run_benchmark` from build.yml 👍

Removed repeated code from build.yml.

**Comment by alexmisk** [comment]
> Yeah, it was supposed to embed PR number (or URL) in the commit title but it seems not working correctly.. I absolutely want more information in gh-pages commits

Will work on it in the coming days!

**Comment by alexmisk** [comment]
> Yeah, it was supposed to embed PR number (or URL) in the commit title but it seems not working correctly.. I absolutely want more information in gh-pages commits

Looks like I've found a relatively consice way to add a PR number to commit messages via Github Actions.

---
**Tl;DR:** 
Now the build.yml triggers when master branch is updated (both with merging a PR or a branch) and sets a commit message to:
- "docs: Update docs for #<PR_number>" when merging a PR 
- "docs: Update docs" when merging a branch
---

This block uses third-party script to get a PR number associated with commit that triggered the workflow run:
```
     - name: Get PR number
        uses: jwalton/gh-find-current-pr@v1
        id: findPr
        with:
          github-token: ${{ secrets.GITHUB_TOKEN }}
```

This one is used to conditionally set a commit message:

```
     - name: Set commit message
        run: |
          if [ -z "$PR" ]
          then
            echo "COMMIT_MESSAGE=docs: Update docs" >> $GITHUB_ENV
          else
            echo "COMMIT_MESSAGE=docs: Update docs for #$PR" >> $GITHUB_ENV
          fi
        env:
          PR: ${{ steps.findPr.outputs.pr }}
```

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      - '!master'
+
+jobs:
+
+  test:
+    name: Run tests with Python [${{ matrix.python-version }}]
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.6, 3.7, 3.8, 3.9-dev, pypy3]
```
Comment (yukinarit) [inline .github/workflows/test.yml:20]: Can we also add this to the matrix?

```yaml
os: [ubuntu-latest, macos-latest, windows-latest]
```

Perhaps this block is going to be like this? 🤔 

```yaml
    runs-on: ${{ matrix.os }}
    strategy:
      matrix:
        python-version: [3.6, 3.7, 3.8, 3.9-dev, pypy3]
        os: [ubuntu-latest, macos-latest, windows-latest]
```

**Comment by yukinarit** [comment]
LGTM for `build.yml` 🎉 👍

File: .github/workflows/test.yml (hunk 1 above)
Comment (alexmisk) [reply .github/workflows/test.yml:20]: Yep. Added MaOS and Windows to test matrix in separate commit. Tests on all platforms are passing btw ;-) Total workflow time now is ~8 min.

**Comment by yukinarit** [comment]
@alexmisk 
Oops Can you remove `appveyor.yml` also?

**Comment by alexmisk** [comment]
> Oops Can you remove `appveyor.yml` also?
Done!

I also realised that we need to run tests on master branch also for proper functioning of status badges :-) So I've edited test.yml a bit.

Old badges removed and new added to README.md

**Review by yukinarit — APPROVED** [review]
ALL LGTM!

**Comment by yukinarit** [comment]
@alexmisk 
Merged. Thank you so much! 😃

**Comment by yukinarit** [comment]
@alexmisk 

There is an extra `[`? 🤔 

image

**Comment by alexmisk** [comment]
> @alexmisk
> 
> There is an extra `[`? 🤔
> 
> image

Fixed with additional PR :-)
